src/dos/cdrom_image.cpp

```diff
@@ -594,34 +911,44 @@ bool CDROM_Interface_Image::AddTrack(Track &curr, int &shift, int prestart, int
 		tracks.push_back(curr);
 		return true;
 	}
-	
+
 	Track &prev = *(tracks.end() - 1);
 	
 	// current track consumes data from the same file as the previous
 	if (prev.file == curr.file) {
 		curr.start += shift;
-		prev.length = curr.start + totalPregap - prev.start - skip;
-		curr.skip += prev.skip + prev.length * prev.sectorSize + skip * curr.sectorSize;		
+		if (!prev.length) {
+			prev.length = curr.start + totalPregap - prev.start - skip;
+		}
+		curr.skip += prev.skip + prev.length * prev.sectorSize + skip * curr.sectorSize;
 		totalPregap += currPregap;
 		curr.start += totalPregap;
 	// current track uses a different file as the previous track
 	} else {
-		int tmp = prev.file->getLength() - prev.skip;
-		prev.length = tmp / prev.sectorSize;
-		if (tmp % prev.sectorSize != 0) prev.length++; // padding
-		
+		if (!prev.length) {
+			int tmp = prev.file->getLength() - prev.skip;
+			prev.length = tmp / prev.sectorSize;
+			if (tmp % prev.sectorSize != 0) prev.length++; // padding
+		}
 		curr.start += prev.start + prev.length + currPregap;
 		curr.skip = skip * curr.sectorSize;
 		shift += prev.start + prev.length;
 		totalPregap = currPregap;
 	}
-	
+		
+#ifdef DEBUG
+	LOG_MSG("%s CDROM: AddTrack cur.start=%d cur.len=%d cur.start+len=%d | prev.start=%d prev.len=%d prev.start+len=%d",
+	        get_time(),
+	        curr.start, curr.length, curr.start + curr.length,
+	        prev.start, prev.length, prev.start + prev.length);
```

Comment:
Tabs got mixed with spaces (but in this case it actually looks ok). Up to you if you want to replace it with tabs, if this wasn't intentional.

Reply:
Yes, this is intentional and how I've tried to line up sub-lines.
```
<tab><tab><tab>First Line Pre-words (x<- position to line-up sub-lines
<tab><tab><tab><..... spaces .....  >Second line
<tab><tab><tab><..... spaces .....  >Third line );
```
Using spaces after the same number of prior-tabs keeps the text alignment regardless of tab-width.  I've made all the LOG_MSG's consistent with this now.